data-prepper-plugins/opensearch/src/main/java/com/amazon/dataprepper/plugins/sink/opensearch/index/IndexManager.java

```diff
@@ -34,14 +43,100 @@
     protected RestHighLevelClient restHighLevelClient;
     protected OpenSearchSinkConfiguration openSearchSinkConfiguration;
     protected IsmPolicyManagementStrategy ismPolicyManagementStrategy;
+    protected String indexPrefix;
 
     private static final Logger LOG = LoggerFactory.getLogger(IndexManager.class);
 
+    //For matching a string that begins with a "%{" and ends with a "}".
+    //For a string like "data-prepper-%{yyyy-MM-dd}", "%{yyyy-MM-dd}" is matched.
+    private final static String TIME_PATTERN_REGULAR_EXPRESSION  = "%\\{.*?\\}";
+
+    //For matching a string enclosed by "%{" and "}".
+    //For a string like "data-prepper-%{yyyy-MM}", "yyyy-MM" is matched.
+    private final static String TIME_PATTERN_INTERNAL_EXTRACTOR_REGULAR_EXPRESSION  = "%\\{(.*?)\\}";
+
+    private Optional<DateTimeFormatter> indexTimeSuffixFormatter;
+    private final static ZoneId UTC_ZONE_ID = ZoneId.of(TimeZone.getTimeZone("UTC").getID());
+
     protected IndexManager(final RestHighLevelClient restHighLevelClient, final OpenSearchSinkConfiguration openSearchSinkConfiguration){
         checkNotNull(restHighLevelClient);
         checkNotNull(openSearchSinkConfiguration);
         this.restHighLevelClient = restHighLevelClient;
         this.openSearchSinkConfiguration = openSearchSinkConfiguration;
+        initializeIndexPrefixAndSuffix();
+    }
+
+    private void initializeIndexPrefixAndSuffix(){
+        final String indexAliasFromConfig = openSearchSinkConfiguration.getIndexConfiguration().getIndexAlias();
+
+        final Pattern pattern = Pattern.compile(TIME_PATTERN_INTERNAL_EXTRACTOR_REGULAR_EXPRESSION);
+        final Matcher matcher = pattern.matcher(indexAliasFromConfig);
+        if (matcher.find()) {
+            if (matcher.groupCount() > 1) {
```

Comment:
This will always return 1 because your matcher pattern only contains 1 group. You would need something like this to count the number of matches:
```
while (matcher.find()) {
    matchCount++;
}
```

Reply:
Good catch! Addressed!

```
        if (timePatternMatcher.find()) {
            final String timePattern = timePatternMatcher.group(1);
            if (timePatternMatcher.find()) { // check if there is a one more match.
                throw new IllegalArgumentException("An index only allows one date-time pattern.");
            }
            ....

        }
```